service/scheduler.go

```diff
@@ -51,6 +78,68 @@ func (s *Scheduler) scheduleWorkflows(db *gorm.DB) {
 	}
 }
 
+func (s *Scheduler) deleteWorkflows(
+	client *orchard.OrchardRestClient,
+	workflowIDs []string,
+	statuses map[string]string,
+) map[string]string {
+	updatedStatuses := statuses
+	for _, orchardID := range workflowIDs {
+		// delete created workflows
+		if err := client.Delete(orchardID); err != nil {
+			fmt.Printf("[error] error deleting workflow: %s\n", err)
+			updatedStatuses[orchardID] = DeleteFailed.ToString()
+		} else {
+			updatedStatuses[orchardID] = Deleted.ToString()
+		}
+	}
+	return updatedStatuses
+}
+
+func (s *Scheduler) cancelWorkflows(
+	client *orchard.OrchardRestClient,
+	statuses map[string]string,
+) map[string]string {
+	updatedStatuses := statuses
+	for orchardID, status := range statuses {
+		if status == Activated.ToString() {
+			// cancel activated workflows
+			if err := client.Cancel(orchardID); err != nil {
+				fmt.Printf("[error] error canceling workflow: %s\n", err)
+				updatedStatuses[orchardID] = CancelFailed.ToString()
+			} else {
+				updatedStatuses[orchardID] = Canceled.ToString()
+			}
+		}
+	}
+	return updatedStatuses
+}
+
+func (s *Scheduler) createActivateWorkflow(
+	client *orchard.OrchardRestClient,
+	wf table.Workflow,
+) map[string]string {
+	statuses := make(map[string]string)
+	createdIDs, err := client.Create(wf)
+	if err != nil {
+		fmt.Printf("[error] error creating workflow: %s\n", err)
+		notifyOwner(wf, err)
+		statuses = s.deleteWorkflows(client, createdIDs, statuses)
+	} else {
+		for _, createdID := range createdIDs {
+			err = client.Activate(createdID)
+			if err != nil {
+				fmt.Printf("[error] error activating workflow: %s\n", err)
+				notifyOwner(wf, err)
+				statuses = s.cancelWorkflows(client, statuses)
```

Comment:
you can break at this point already, no?

Reply:
@realstraw yes, changed